### PR TITLE
Frog PR 🐸 

### PR DESCRIPTION
**Pull request by lisyce**
If I did this part wrong I'm sorry, git is hard.

### CODE REVIEW

File: src/main/java/Commands/Frog/FrogFactCommand.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    private List<String> allFacts = new ArrayList<>();
+
+    public FrogFactCommand() {
+        super("frogfact", CommandPermission.EVERYONE, "Get a random frog fact");
+    }
+
+    @Override
+    public void runCommand(CommandContext ctx) {
+
+        File frogFactList = new File("src/main/resources/frog-facts.txt");
```
Comment (1Revenger1) [inline src/main/java/Commands/Frog/FrogFactCommand.java:24]: Can this be moved into the try/catch?

File: src/main/java/Commands/Frog/SenderReceiverCommand.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ * Useful for commands where the bot pings a sender and receiver such as $bonk and $validate
+ */
+public abstract class SenderReceiverCommand extends ServerCommand {
+
+    private final String noSpecifiedRecipientMsg;
+    private final String successfulMsg;
+
+    public SenderReceiverCommand(String keyword, CommandPermission cmdPerm, String description, String noSpecifiedRecipientMsg, String successfulMsg) {
+        super(keyword, cmdPerm, description);
+        this.noSpecifiedRecipientMsg = noSpecifiedRecipientMsg;
```
Comment (1Revenger1) [inline src/main/java/Commands/Frog/SenderReceiverCommand.java:18]: Be cool if we could pass in an array of success/failure values which could be used randomly.

**Review by 1Revenger1 — CHANGES_REQUESTED** [review]

File: src/main/java/Commands/Frog/BonkCommand.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
+package Commands.Frog;
+
+public class BonkCommand extends SenderReceiverCommand {
+
+    public BonkCommand() {
+        super("bonk", CommandPermission.EVERYONE,
+                """
+                        Bonks another user
+                        Usage: bonk <user tag>
```
Comment (SnakeJ419) [inline src/main/java/Commands/Frog/BonkCommand.java:9]: if you put %{p} infront of the keyword it automagically gets replaced with the correct prefix

**Review by 1Revenger1 — DISMISSED** [review]

File: src/main/java/Commands/Frog/ValidateCommand.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
+package Commands.Frog;
+
+
+public class ValidateCommand extends SenderReceiverCommand {
+
+    public ValidateCommand() {
+        super("validate", CommandPermission.EVERYONE,
+                """
+                        Remind another user that they matter to you
+                        Usage: validate <user tag>
```
Comment (SnakeJ419) [inline src/main/java/Commands/Frog/ValidateCommand.java:10]: same thing here, %{p}validate will automagically put the right prefix infront

**Review by SnakeJ419 — APPROVED** [review]
Very simple change, looks good

**Review by SnakeJ419 — APPROVED** [review]
👍

**Review by 1Revenger1 — APPROVED** [review]

**Pull request merged**